tests/testsuite/update.rs

```diff
@@ -1637,3 +1639,429 @@ fn update_with_missing_feature() {
+        .file(
+            "bar/Cargo.toml",
+            r#"
+                [package]
+                name = "bar"
+                version = "0.0.1"
+                edition = "2015"
+                authors = []
+
+                [dependencies]
+                multiple = "2.0"
+            "#,
+        )
+        .file("bar/src/lib.rs", "")
+        .build();
+
+    p.cargo("generate-lockfile").run();
+
+    Package::new("compatible", "1.0.1").publish();
+    Package::new("incompatible", "1.0.1").publish();
+    Package::new("pinned", "1.0.1").publish();
+    Package::new("less-than", "1.0.1").publish();
+    Package::new("renamed-from", "1.0.1").publish();
+    Package::new("ws", "1.0.1").publish();
+    Package::new("multiple", "1.0.1").publish();
+    Package::new("multiple", "2.0.1").publish();
+    Package::new("alternative-1", "1.0.1")
+        .alternative(true)
+        .publish();
+    Package::new("alternative-2", "1.0.1")
+        .alternative(true)
+        .publish();
+
+    Package::new("incompatible", "2.0.0").publish();
+    Package::new("pinned", "2.0.0").publish();
+    Package::new("less-than", "2.0.0").publish();
+    Package::new("renamed-from", "2.0.0").publish();
+    Package::new("pre-release", "2.0.0-alpha").publish();
+    Package::new("yanked", "2.0.0").yanked(true).publish();
+    Package::new("ws", "2.0.0").publish();
+    Package::new("multiple", "3.0.0").publish();
+    Package::new("alternative-1", "2.0.0")
+        .alternative(true)
+        .publish();
+    Package::new("alternative-2", "2.0.0")
+        .alternative(true)
+        .publish();
+
+    p.cargo("update -Zunstable-options --breaking")
+        .masquerade_as_nightly_cargo(&["update-breaking"])
+        .with_stderr_unordered(
+            "\
+[UPDATING] `alternative` index
+[UPGRADING] alternative-1 ^1.0 -> v2.0.0
+[UPGRADING] alternative-2 ^1.0 -> v2.0.0
+[UPDATING] `dummy-registry` index
+[UPGRADING] incompatible ^1.0 -> v2.0.0
+[UPGRADING] multiple ^1.0 -> v3.0.0
+[UPGRADING] ws ^1.0 -> v2.0.0
+[UPGRADING] multiple ^2.0 -> v3.0.0
+[LOCKING] 5 packages to latest compatible versions
+[UPDATING] alternative-1 v1.0.0 (registry `alternative`) -> v2.0.0
+[UPDATING] alternative-2 v1.0.0 (registry `alternative`) -> v2.0.0
+[UPDATING] incompatible v1.0.0 -> v2.0.0
+[REMOVING] multiple v1.0.0
+[REMOVING] multiple v2.0.0
+[ADDING] multiple v3.0.0
+[UPDATING] ws v1.0.0 -> v2.0.0
+[NOTE] pass `--verbose` to see 4 unchanged dependencies behind latest
+",
+        )
+        .run();
+
+    let root_manifest = p.read_file("Cargo.toml");
+    assert_e2e().eq(
+        &root_manifest,
+        str![[r#"
+
+                # Check if formatting is preserved
+
+                [workspace]
+                members  =  ["foo", "bar"]
+
+                [workspace.dependencies]
+                ws  = "2.0"
+            
+"#]],
+    );
+
+    let crate_manifest = p.read_file("foo/Cargo.toml");
+
+    // FIXME: Not sure how to preserve all formatting here.
```

Comment:
Tracking this FIXME

Reply:
I believe `cargo add` is able to preserve formatting in these cases, so I don't think this is a problem with `toml_edit` but how we are using it.

Reply:
The distinction is that this code is overwriting the toml table
```rust
*dep_item = dep.to_toml(manifest_path);
```
while cargo-add updates it
```rust
dep.update_toml(&crate_root, &mut dep_key, dep_item);
```

Reply:
Excellent! Fixed.